units/crons.go

```diff
@@ -320,24 +320,37 @@ func PopulateIdleHostJobs(env evergreen.Environment) amboy.QueueOperation {
 
 		catcher := grip.NewBasicCatcher()
 		ts := util.RoundPartOfHour(1).Format(tsFormat)
-		hosts, err := host.AllIdleEphemeral()
+		distroHosts, err := host.IdleEphemeralGroupedByDistroId(true)
 		catcher.Add(err)
-		grip.Warning(message.WrapError(err, message.Fields{
-			"cron":      idleHostJobName,
-			"operation": "background task creation",
-			"hosts":     hosts,
-			"impact":    "idle hosts termination",
-		}))
+
+		var hostsToTerminate []host.Host
+		for _, info := range distroHosts {
+			nIdleHosts := len(info.Hosts)
+			distro, err := distro.FindOne(distro.ById(info.DistroID))
+			if err != nil {
+				catcher.Add(err)
+			}
+			minimumHosts := distro.PlannerSettings.MinimumHosts
+
+			if nIdleHosts <= minimumHosts {
+				continue
+			}
+
+			nHostsToTerminate := nIdleHosts - minimumHosts
```

Comment:
This doesn't take into account non-idle hosts right? I would think that we should subtract the min hosts number from the total number of hosts

Reply:
Done.